### PR TITLE
Sign up

### PR DESCRIPTION
**Pull request by leherreras**
As a user I should be able to sign up so my account is created in Target MVD

### CODE REVIEW

File: config/application.rb [hunk 1]
```diff
@@ -11,15 +11,21 @@
 module RooLC
   # Root config for owr applications
   class Application < Rails::Application
-    # Initialize configuration defaults for originally generated Rails version.
+    # Initialize configuration duser_spec.rbefaults for originally generated Rails version.
```
Comment (enzofab91) [inline config/application.rb]:
```suggestion
    # Initialize configuration user_spec.rb defaults for originally generated Rails version.
```

File: spec/requests/api/v1/registrations_spec.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  describe 'POST /create' do
+    context 'with valid parameters' do
+      it 'creates a new User' do
+        expect do
+          post '/api/v1', params: params
+        end.to change(User, :count).by(1)
+      end
+
+      it 'successful response' do
+        post '/api/v1', params: params, as: :json
```
Comment (enzofab91) [inline spec/requests/api/v1/registrations_spec.rb]: You can use `subject`.

See https://relishapp.com/rspec/rspec-core/docs/subject/explicit-subject

**Review by enzofab91 — APPROVED** [review]
Good job!

Left some comments. Check Rubocop results.
It seems that this PR has things related to another feature. Be careful in the next ones.

File: config/application.rb (hunk 1 above)
Comment (leherreras) [reply config/application.rb]: the suggest was apply in one commit thanks

File: spec/requests/api/v1/registrations_spec.rb (hunk 2 above)
Comment (leherreras) [reply spec/requests/api/v1/registrations_spec.rb]: the suggest was apply in one commit thanks

File: spec/requests/api/v1/registrations_spec.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  let(:params) do
+    { email: email, gender: gender, password: password, password_confirmation: password }
+  end
+
+  describe 'POST /create' do
+    context 'with valid parameters' do
+      it 'creates a new User' do
+        expect do
+          subject
+        end.to change(User, :count).by(1)
```
Comment (santir489) [inline spec/requests/api/v1/registrations_spec.rb]: we can write this in one line right? like the others

File: spec/requests/api/v1/registrations_spec.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Api::V1::Registrations', type: :request do
+  subject { post user_registration_path, params: params, as: :json }
+
+  let(:email) { Faker::Internet.unique.email }
```
Comment (santir489) [inline spec/requests/api/v1/registrations_spec.rb]: Here you can use person attributes generated with the factory. WDYT?

File: spec/requests/api/v1/registrations_spec.rb (hunk 3 above)
Comment (leherreras) [reply spec/requests/api/v1/registrations_spec.rb]: Thanks, in the new commit 
`expect { subject }.to change(User, :count).by(1)`

File: spec/requests/api/v1/registrations_spec.rb (hunk 4 above)
Comment (leherreras) [reply spec/requests/api/v1/registrations_spec.rb]: That is a good ides, that was implement in the new commit

File: app/controllers/api/v1/registrations_controller.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    # Override the registration class to accept the gender parameter
```
Comment (sebastiancaraballo) [inline app/controllers/api/v1/registrations_controller.rb:5]: Avoid comments

File: config/locales/en.yml [hunk 6]
```diff
@@ -1,2 +1,14 @@
...
+  api:
+    errors:
+      invalid_content_type: Invalid content type header
+      missing_param: A required param is missing
+      not_found: "Couldn't find the record"
+      target_limit: Users target limit is %{Target::USER_MAX_TARGET_LIMIT}
+      match_same_user: "Can't match a target with yourself"
+      uniqueness_conversation: "Can't create conversation for the same match twice"
+      message_empty: "The message can't be empty"
+      facebook_not_authorized: Not Authorized
```
Comment (sebastiancaraballo) [inline config/locales/en.yml:13]: Are these being used? 🤔

File: spec/requests/api/v1/registrations_spec.rb [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+      it 'valid response with the params' do
+        subject
+        expect(json[:status]).to eq('success')
+        expect(json[:data][:email]).to eq(user.email)
+        expect(json[:data][:uid]).to eq(user.email)
+        expect(json[:data][:gender]).to eq(user.gender)
+      end
+    end
+
+    context 'with invalid parameters' do
```
Comment (sebastiancaraballo) [inline spec/requests/api/v1/registrations_spec.rb:34]: Maybe add an example `it returns an unsuccessful response`

**Review by sebastiancaraballo — APPROVED** [review]
Nice

File: app/controllers/api/v1/registrations_controller.rb (hunk 5 above)
Comment (leherreras) [reply app/controllers/api/v1/registrations_controller.rb:5]: this comments are suggestions of rubocop to know about the work in class

File: spec/requests/api/v1/registrations_spec.rb (hunk 7 above)
Comment (leherreras) [reply spec/requests/api/v1/registrations_spec.rb:34]: ok, thanks

File: config/locales/en.yml (hunk 6 above)
Comment (leherreras) [reply config/locales/en.yml:13]: this are the comments to test another functionalities

File: app/controllers/api/v1/registrations_controller.rb (hunk 5 above)
Comment (sebastiancaraballo) [reply app/controllers/api/v1/registrations_controller.rb:5]: Well, that's a nice cop to exclude then. Take a look at [this](https://github.com/rootstrap/tech-guides/tree/master/ruby#comments).

**Pull request merged**